modules/build/src/main/scala/scala/build/options/ScalaOptions.scala

```diff
@@ -12,16 +12,16 @@ final case class ScalaOptions(
   scalacOptions: Seq[String] = Nil,
   extraScalaVersions: Set[String] = Set.empty,
   compilerPlugins: Seq[Positioned[AnyDependency]] = Nil,
-  platform: Option[Platform] = None,
-  extraPlatforms: Set[Platform] = Set.empty
+  platform: Option[Positioned[Platform]] = None,
+  extraPlatforms: Map[Platform, Positioned[Unit]] = Map.empty
 ) {
   def normalize: ScalaOptions = {
     var opt = this
     for (sv <- opt.scalaVersion if opt.extraScalaVersions.contains(sv))
       opt = opt.copy(
         extraScalaVersions = opt.extraScalaVersions - sv
       )
-    for (pf <- opt.platform if opt.extraPlatforms.contains(pf))
+    for (pf <- opt.platform.map(_.value) if opt.extraPlatforms.keys.toList.contains(pf))
```

Comment:
No need to convert the key set to a list to only call `contains` on it:
```suggestion
    for (pf <- opt.platform.map(_.value) if opt.extraPlatforms.keySet.contains(pf))
```